src/main/java/org/prebid/server/geolocation/GeoLite.java

```diff
@@ -0,0 +1,69 @@
+package org.prebid.server.geolocation;
+
+import com.maxmind.db.Reader;
+import com.maxmind.geoip2.DatabaseReader;
+import com.maxmind.geoip2.exception.GeoIp2Exception;
+import com.maxmind.geoip2.model.CountryResponse;
+import io.vertx.core.Future;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.execution.Timeout;
+import org.prebid.server.geolocation.model.GeoInfo;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.Objects;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Default implementation of {@link GeoLocationService}
+ */
+public class GeoLite implements GeoLocationService {
+
+    private final DatabaseReader databaseReader;
+
+    private GeoLite(DatabaseReader databaseReader) {
+        this.databaseReader = Objects.requireNonNull(databaseReader);
+    }
+
+    public static GeoLite create(String dbArchive) {
+        try (TarArchiveInputStream tarInput =
+                     new TarArchiveInputStream(new GZIPInputStream(
+                             GeoLite.class.getClassLoader().getResourceAsStream(dbArchive)))) {
+
+            TarArchiveEntry currentEntry;
+            while ((currentEntry = tarInput.getNextTarEntry()) != null) {
+                if (currentEntry.getName().contains(".mmdb")) {
+                    break;
```

Comment:
What will happen if the archive missing ".mmdb" file or contains multiple files with such extension?